src/twisted/test/test_compat.py

```diff
@@ -921,3 +921,32 @@ def test_bytesReprPrefix(self):
         ``b`` to the returned repr on both Python 2 and 3.
         """
         self.assertEqual(_bytesRepr(b'\x00'), "b'\\x00'")
+
+
+
+class GetAsyncParamTests(unittest.SynchronousTestCase):
+    """
+    Tests for L{get_async_param}
+    """
+
+    def test_get_async_param(self):
+        """
+        L{twisted.python.compat.get_async_param}
+        """
+        self.assertEqual(get_async_param(isAsync=False), False)
+        self.assertEqual(get_async_param(isAsync=True), True)
+        self.assertEqual(
+            get_async_param(isAsync=None, **{'async': False}), False)
+        self.assertEqual(
+            get_async_param(isAsync=None, **{'async': True}), True)
+        self.assertRaises(TypeError, get_async_param, False, {'async': False})
+
+
+    def test_get_async_param_deprecation(self):
```

Comment:
The test should be for the public API itself, and not for this (what should be) private code.